test/wh_test_clientserver.c

```diff
@@ -624,267 +623,485 @@ int whTest_ClientServerSequential(void)
+               ret, gid, gaccess, gflags, glen, glabel);
+#endif
+
+        /* Ensure metadata matches that of the object we just wrote */
+        WH_TEST_ASSERT_RETURN(server_rc == WH_ERROR_OK);
+        WH_TEST_ASSERT_RETURN(gid == id);
+
+        memset(recv_buffer, 0, sizeof(recv_buffer));
+        WH_TEST_RETURN_ON_FAIL(ret = wh_Client_NvmRead(client, id, 0, glen,
+                                                       &server_rc, &rlen,
+                                                       (uint8_t*)recv_buffer));
+
+#if defined(WH_CFG_TEST_VERBOSE)
+        printf(
+            "Client NvmRead:%d, server_rc:%d id:%u, len:%u data:%s\n",
+            ret, server_rc, gid, rlen, recv_buffer);
+#endif
+
+        /* Ensure data and size of response object matches that of the written
+         * object */
+        WH_TEST_ASSERT_RETURN(server_rc == WH_ERROR_OK);
+        WH_TEST_ASSERT_RETURN(rlen == len);
+        WH_TEST_ASSERT_RETURN(0 == memcmp(send_buffer, recv_buffer, len));
+    }
+
+    whNvmAccess list_access = WOLFHSM_NVM_ACCESS_ANY;
+    whNvmFlags  list_flags  = WOLFHSM_NVM_FLAGS_ANY;
+    whNvmId     list_id     = 0;
+    whNvmId     list_count  = 0;
+    do {
+        WH_TEST_RETURN_ON_FAIL(
+            ret = wh_Client_NvmList(client, list_access, list_flags, list_id,
+                              &server_rc, &list_count, &list_id));
+#if defined(WH_CFG_TEST_VERBOSE)
+        printf("Client NvmList:%d, server_rc:%d count:%u id:%u\n", ret,
+               server_rc, list_count, list_id);
+#endif
+
+        if (list_count > 0) {
+            /* ensure list_id contains ID of object written, and list_count
+             * shows remaining items in list */
+            WH_TEST_ASSERT_RETURN(list_id == 20 + (5 - list_count));
+
+            WH_TEST_RETURN_ON_FAIL(
+                ret = wh_Client_NvmDestroyObjects(client, 1, &list_id, 0, NULL,
+                                                  &server_rc));
+
+#if defined(WH_CFG_TEST_VERBOSE)
+            printf("Client NvmDestroyObjects:%d, server_rc:%d for "
+                   "id:%u with count:%u\n",
+                   ret, server_rc, list_id, list_count);
+#endif
+            WH_TEST_ASSERT_RETURN(server_rc == WH_ERROR_OK);
+
+            /* Ensure object was destroyed and no longer exists */
+            WH_TEST_RETURN_ON_FAIL(ret = wh_Client_NvmGetMetadata(client, list_id, &server_rc, NULL, NULL, NULL, NULL, 0, NULL));
+            WH_TEST_ASSERT_RETURN(WH_ERROR_NOTFOUND == server_rc);
+
+#if defined(WH_CFG_TEST_VERBOSE)
+            printf("Client NvmGetMetadata:%d, server_rc:%d count:%u id:%u\n",
+                   ret, server_rc, list_count, list_id);
+#endif
+
+            list_id = 0;
         }
+    } while (list_count > 0);
+
+
+    WH_TEST_RETURN_ON_FAIL(wh_Client_NvmGetAvailable(
+        client, &server_rc, &avail_size, &avail_objects, &reclaim_size,
+        &reclaim_objects));
+    WH_TEST_ASSERT_RETURN(avail_objects == NF_OBJECT_COUNT);
```

Comment:
Grumble.  This should have been the global config WOLFHSM_NUM_NVMOBJECTS  (in wh_common.h) and NF_OBJECT_COUNT should have been set to this or replaced.  Fix if you want.

Reply:
fixed